airflow/operators/singularity_operator.py

```diff
@@ -0,0 +1,178 @@
+
+    @apply_defaults
+    def __init__(
+            self,
+            image,
+            api_version=None,
+            command=None,
+            start_command=None,
+            environment=None,
+            pull_folder=None,
+            force_pull=False,
+            volumes=None,
+            options = None,
+            working_dir=None,
+            auto_remove=False,
+            *args,
+            **kwargs):
+
+        super(SingularityOperator, self).__init__(*args, **kwargs)
+        self.api_version = api_version
+        self.auto_remove = auto_remove
+        self.command = command
+        self.start_command = start_command
+        self.environment = environment or {}
+        self.force_pull = force_pull
+        self.image = image
+        self.instance = None
+        self.options = options or []
+        self.volumes = volumes or []
+        self.working_dir = working_dir
+        self.cli = None
+        self.container = None
+
+    def execute(self, context):
+        self.log.info('Preparing Singularity container %s', self.image)
+        self.cli = Client
+
+        if self.command == None:
+            raise AirflowException('You must define a command.')
+
+        # Pull the container if asked, and ensure not a binary file
+        if self.force_pull and not os.path.exists(self.image):
+            self.log.info('Pulling container %s', self.image)
+            image, lines = self.cli.pull(self.image, stream=True) 
+            for line in lines:
+                self.log.info(line)
+
+            # Move the container to where it's desired
+            if pull_folder != None:
+                self.image = os.path.join(pull_folder, os.path.basename(image))
+                shutil.move(image, self.image)
+
+        with TemporaryDirectory(prefix='airflowtmp') as host_tmp_dir:
+
+            # Prepare list of binds
+            for bind in self.volumes:
+                self.options = self.options + ['--bind', bind]
+
+            # Does the user want a custom working directory?
+            if self.working_dir != None:
+                self.options = self.options + ['--workdir', self.working_dir]
+ 
+            # Export environment before instance is run
+            for enkey, envar in self.environment.items():
+                self.log.debug('Exporting %s=%s' %(envar, enkey))
+                os.putenv(enkey, envar)
+                os.environ[enkey] = envar
+
+            # Create a container instance
+            print(self.options)
+            self.instance = self.cli.instance(self.image, 
+                                              options=self.options,
+                                              args=self.start_command)
+
+            print(self.instance.cmd)
+            self.log.info('Created instance %s from %s' %(self.instance, 
+                                                          self.image))
+
+            self.log.debug('Running command %s' % self.get_command())
```

Comment:
```suggestion
            self.log.debug('Running command %s', self.get_command())
```
Reference: https://github.com/apache/airflow/pull/4804